src/SupabaseClient.ts

```diff
@@ -164,37 +164,29 @@ export default class SupabaseClient {
    *
    * @param subscription The subscription you want to remove.
    */
-  removeSubscription(subscription: RealtimeSubscription): Promise<
+  async removeSubscription(subscription: RealtimeSubscription): Promise<
     | {
         data: { openSubscriptions: number }
         error: null
       }
     | { error: Error }
   > {
-    return new Promise(async (resolve, reject) => {
       const { error } = await this._closeSubscription(subscription)
-
       if (error) {
-        return reject({ error })
+        return { error }
```

Comment:
@w3b6x9 is the function supposed to throw on error? If so this should become `throw { error }` and L172 should be removed, otherwise this is correct.